test/vre/chapter36/config/veteranInformation.unit.spec.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { expect } from 'chai';
+import sinon from 'sinon';
+import { mount } from 'enzyme';
+
+import { DefinitionTester, fillData } from '../../../util/schemaform-utils.jsx';
+import formConfig from '../../../../src/js/vre/chapter36/config/form.js';
+
+describe('VRE chapter 36 veteran information', () => {
+  const { schema, uiSchema } = formConfig.chapters.veteranInformation.pages.veteranInformation;
+  it('renders veteran information fields', () => {
+    const form = mount(
+      <DefinitionTester
+        definitions={formConfig.defaultDefinitions}
+        schema={schema}
+        data={{
+          'view:isVeteran': true
+        }}
+        formData={{}}
+        uiSchema={uiSchema}/>
+    );
+
+    expect(form.find('input').length).to.equal(6);
+    expect(form.find('select').length).to.equal(3);
+  });
+
+  it('renders veteran date of death field when view:isVeteran is false', () => {
+    const form = mount(
+      <DefinitionTester
+        definitions={formConfig.defaultDefinitions}
+        schema={schema}
+        data={{
+          'view:isVeteran': false
+        }}
+        formData={{}}
+        uiSchema={uiSchema}/>
+    );
+
+    expect(form.find('input').length).to.equal(7);
+    expect(form.find('select').length).to.equal(5);
+  });
+
+  it('does not submit without required veteran information', () => {
+    const onSubmit = sinon.spy();
+    const form = mount(
+      <DefinitionTester
+        definitions={formConfig.defaultDefinitions}
+        schema={schema}
+        data={{
+          'view:isVeteran': false
+        }}
+        formData={{}}
+        onSubmit={onSubmit}
+        uiSchema={uiSchema}/>
+    );
+    form.find('form').simulate('submit');
+    expect(form.find('.usa-input-error').length).to.equal(4);
+
+    expect(onSubmit.called).to.be.false;
+  });
+
+
+  it('submits with only required information', () => {
+    const onSubmit = sinon.spy();
+    const form = mount(
+      <DefinitionTester
+        definitions={formConfig.defaultDefinitions}
+        schema={schema}
+        data={{
+          'view:isVeteran': false
+        }}
+        formData={{}}
+        onSubmit={onSubmit}
+        uiSchema={uiSchema}/>
+    );
+    fillData(form, 'input#root_veteranFullName_first', 'test');
+    fillData(form, 'input#root_veteranFullName_last', 'test2');
+    fillData(form, 'select#root_veteranDateOfBirthMonth', '1');
+    fillData(form, 'select#root_veteranDateOfBirthDay', '3');
+    fillData(form, 'input#root_veteranDateOfBirthYear', '1950');
+    fillData(form, 'input#root_veteranSocialSecurityNumber', '424242424');
+    fillData(form, '[id="root_view:veteranDateOfDeathMIAPOW_veteranDateOfDeathMIAPOWMonth"]', '1');
```

Comment:
Enzyme did not like this `:`. I tried to escape it with `\\` and it blew up.

Reply:
Does escaping use one `\` or two?

Reply:
It blew up with `\` and gave me a parsing error when I used `\\`. It displayed the properly escape CSS query with the second error.

Reply:
Thanks for the helpful context. It looks like they have an open issue for this (1218).

Reply:
I should get into the habit of searching / filing issues on open source projects.